l1-contracts/contracts/state-transition/chain-deps/facets/Mailbox.sol

```diff
@@ -197,78 +191,95 @@ contract MailboxFacet is Base, IMailbox {
         uint256 _l2GasPerPubdataByteLimit,
         bytes[] calldata _factoryDeps,
         address _refundRecipient
-    ) external payable nonReentrant returns (bytes32 canonicalTxHash) {
+    ) external payable returns (bytes32 canonicalTxHash) {
+        require(s.chainId == ERA_CHAIN_ID, "legacy interface only available for eth base token");
+        canonicalTxHash = _requestL2TransactionSender(
+            BridgehubL2TransactionRequest({
+                sender: msg.sender,
+                contractL2: _contractL2,
+                mintValue: msg.value,
+                l2Value: _l2Value,
+                l2Calldata: new bytes(0),
+                l2GasLimit: _l2GasLimit,
+                l2GasPerPubdataByteLimit: _l2GasPerPubdataByteLimit,
+                l1GasPriceConverted: tx.gasprice,
+                factoryDeps: new bytes[](0),
+                refundRecipient: _refundRecipient
+            }),
+            _calldata,
+            _factoryDeps
+        );
+        IL1SharedBridge(s.baseTokenBridge).bridgehubDepositBaseToken{value: msg.value}(
+            s.chainId,
+            msg.sender,
+            ETH_TOKEN_ADDRESS,
+            msg.value
+        );
+    }
+
+    function _requestL2TransactionSender(
+        BridgehubL2TransactionRequest memory _request,
+        bytes calldata _calldata,
+        bytes[] calldata _factoryDeps
```

Comment:
why do we need these two fields here as these are already included in the `request`?

Reply:
I will remove it from the inner struct.

Reply:
But you end up copying those anyway for calculating the hash of the priority tx, no? 

It would be interesting to see gas benchmarks between this approach and just using `memory` rightaway, since right now it involves code duplication. If the gain is high, then sure it is worth it, but I am wondering whether it is really worth it

Reply:
True, we do need to calculate the hash. Should I just go for memory?